elastic/integrations
# [system testing] Validate assertion for expected number of events

## What does this PR do?

This PR will validate `system testing` through the addition of an assertion for expected number of events

## Checklist

~- [ ] I have verified that all data streams collect metrics or logs.~
~- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).~

## Author's Checklist


- [ ] Wait for the outcome of the `elastic-package` PR [[system testing] Implement assertion for expected number of events](https://github.com/elastic/elastic-package/pull/960#top) since these changes lack meaning outside that context.

## How to test this PR locally

Ensure `elastic-package` is up to date by running 

`make install`

in the `./elastic-package` root directory of an up-to-date clone or synced fork of this repository

In the `integrations` repository, run the following commands from the `./integrations` root directory

```
cd ../httpjson
elastic-package stack down -v
elastic-package build -v
elastic-package stack up --version 8.5.0-SNAPSHOT -v -d
elastic-package test system -v
```

Validate that the output contains:

- Debug entries similar to the following:

```
2022/11/22 20:12:47 DEBUG running test with configuration 'expected-event-count'
2022/11/22 20:12:47 DEBUG setting up service...
...
2022/11/22 20:13:07 DEBUG checking for expected data in data stream...
2022/11/22 20:13:07 DEBUG found 1 hits in logs-httpjson.generic-ep data stream
2022/11/22 20:13:07 DEBUG found 3 events under "parent"
```

(The assertion `found 3 events under "parent"` will only triggered after the related `elastic-package` PR has merged)

- A Test result similar to the following:

    ```│ httpjson │ generic     │ system    │ expected-event-count │ PASS   │ 20.829026099s │```
    
## Related issues

- Relates https://github.com/elastic/elastic-package/issues/821

~## Screenshots~
